From 0f0f5be1b143cb8a483e11229d8d27ab85835a5c Mon Sep 17 00:00:00 2001
From: Juan Manuel Lopez Carrillo <juanmanuellopezcarrillo@gmail.com>
Date: Wed, 8 Jul 2026 02:15:26 +0200
Subject: [PATCH 012/130] dts: opi4a: add 300ps of syscon RX (total ~2.2ns =
 BSP)

With PJ10 the RX came back but marginal (rx_packets crept up one at a time,
0 crc). We add rx-internal-delay-ps=300 on the MAC on top of the PHY 1.9ns
for the total ~2.2ns of the BSP.
---
 arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts b/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
index cdc678c37..543309b70 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
@@ -501,7 +501,10 @@ &gmac1 {
 	phy-handle = <&ext_rgmii_phy>;
 	phy-supply = <&reg_cldo4>;
 	tx-internal-delay-ps = <100>;
-	rx-internal-delay-ps = <0>;
+	/* PHY adds 1.9ns of RXC (rgmii-rxid + PHY rx=1900); add 300ps
+	 * of syscon for the total ~2.2ns of the BSP (RX marginal with only
+	 * 1.9ns: rx_packets crept up one at a time). */
+	rx-internal-delay-ps = <300>;
 	status = "okay";
 };
 
